torchgeo/datasets/utils.py

```diff
@@ -847,3 +851,20 @@ def array_to_tensor(array: np.typing.NDArray[Any]) -> Tensor:
     elif array.dtype == np.uint32:
         array = array.astype(np.int64)
     return torch.tensor(array)
+
+
+def valid_data_footprint_from_datasource(
+    src: rasterio.io.DatasetReader, destination_crs: pyproj.crs.crs.CRS
+) -> Polygon:
+    # Read valid/nodata-mask
+    mask = src.read_masks()
```

Comment:
`read_masks()` returns `C, H, W` shape it has more than one band/mask

probably needs to do something like

```python
if len(masks) == 3:
    mask = np.logical_or(*masks[:, ...])
```

Reply:
We should choose to only use the mask for one spectral band, and it should probably work well, like `mask=src.read_masks(1)`. But in case there is a spatial shift between the bands, passing all of them to `sieve` will (to my understanding) find the valid mask across **all** bands.

Reply:
So I think your proposal is covered by `sieve`?

Reply:
1 spectral band may not be enough, especially in the case of Landsat 7 SLC-off. Would be curious to know how far things shift. If it's only a few pixels, then that's within the margin of error of this method.

Reply:
Looking at this again I see that we indeed need the logical_or to cover shifts between spectral bands. Will push soon!

(My last comment about `sieve` (or `shape`) solving this was not correct)

Reply:
`np.logical_or.reduce(mask, axis=0).astype('uint8')` now accept any number of bands.